.instrumentation_generator/test/instrumentation_generator_test.rb

```diff
@@ -0,0 +1,84 @@
+# test/instrumentation_generator_test.rb
+
+require 'minitest/autorun'
+require 'mocha/minitest'
```

Comment:
I wasn't able to run these tests locally. When I tried to run the file from the root directory using: 
```
bundle exec ruby .instrumentation_generator/test/instrumentation_generator_test.rb
```

I got the following error: 
```
/Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/mocha-1.16.1/lib/mocha/integration/mini_test/adapter.rb:26:in `included': uninitialized constant MiniTest (NameError)

          Mocha::ExpectationErrorFactory.exception_class = ::MiniTest::Assertion
                                                           ^^^^^^^^^^
Did you mean?  Minitest
        from /Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/mocha-1.16.1/lib/mocha/integration/mini_test.rb:50:in `include'
        from /Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/mocha-1.16.1/lib/mocha/integration/mini_test.rb:50:in `activate'
        from /Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/mocha-1.16.1/lib/mocha/minitest.rb:5:in `<top (required)>'
        from <internal:/Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from <internal:/Users/kreopelle/.rbenv/versions/3.3.1/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
        from instrumentation_generator_test.rb:4:in `<main>'
```

It looks like `mocha` hasn't updated its call to `MiniTest::Assertion` to use the `Minitest` spelling. Minitest used to accept both spellings, but now it only accepts `Minitest`. 

Minitest has its own stubbing features that can be used, though you have to use them in a specific test and can't set them in a `setup` method. Here's one of my favorite articles on stubbing in Minitest: https://semaphoreci.com/community/tutorials/mocking-in-ruby-with-minitest 

When I removed `mocha/minitest` and took the `@generator.stubs` method calls out of the `setup` method, I was able to run the tests.

The only error I got was related to calling `puts`:
```
  1) Error:
InstrumentationGeneratorTest#test_update_ci_workflow:
NoMethodError: private method `puts' called for an instance of InstrumentationGenerator
    .instrumentation_generator/test/instrumentation_generator_test.rb:86:in `test_update_ci_workflow'

6 runs, 21 assertions, 0 failures, 1 errors, 0 skips
```